src/lib/journey/callbacks/recaptcha/recaptcha.svelte

```diff
@@ -23,57 +30,79 @@
-          'error-callback': 'frHandleErrorCallback',
-        });
-      } else {
-        window.grecaptcha.render(recaptchaClass, {
-          sitekey: siteKey,
-          callback: window.frHandleCaptcha, // grecaptcha uses different types so passing real function here
-          'expired-callback': window.frHandleExpiredCallback,
-        });
-      }
-    };
+    if (!isV3) {
+      window.frHandleCaptchaError = function handleCaptchaError() {
+        callback?.setResult('');
+        if (recaptchaClass.startsWith('h')) {
+          window.hcaptcha.render(recaptchaClass, {
+            sitekey: siteKey,
+            callback: 'frHandleCaptcha',
+            'expired-callback': 'frHandleExpiredCallback',
+            'chalexpired-callback': 'frHandleExpiredCallback',
+            'error-callback': 'frHandleErrorCallback',
+          });
+        } else {
+          window.grecaptcha.render(recaptchaClass, {
+            sitekey: siteKey,
+            callback: window.frHandleCaptcha, // grecaptcha uses different types so passing real function here
+            'expired-callback': window.frHandleExpiredCallback,
+          });
+        }
+      };
+      window.frHandleCaptcha = function handleCaptchaToken(token: string) {
+        callback?.setResult(token);
+      };
+      window.frHandleExpiredCallback = function handleExpiredCallback() {
+        callback?.setResult('');
+        if (recaptchaClass.startsWith('h')) {
+          window.hcaptcha.render(recaptchaClass, {
+            sitekey: siteKey,
+            callback: 'frHandleCaptcha',
+            'expired-callback': 'frHandleExpiredCallback',
+            'chalexpired-callback': 'frHandleExpiredCallback',
+            'error-callback': 'frHandleErrorCallback',
+          });
+        } else {
+          window.grecaptcha.render(recaptchaClass, {
+            sitekey: siteKey,
+            callback: window.frHandleCaptcha, // grecaptcha uses different types so passing real function here
+            'expired-callback': window.frHandleExpiredCallback,
+          });
+        }
+      };
+    }
   });
+  // defining this outside of the reactive block and guarding it with a isV3 check so it only runs when v3
+  // is defined as true and we have a recaptcha action to assign.
+  function executeCaptcha() {
+    if (isV3 && recaptchaAction.length) {
+      window.grecaptcha.ready(async function () {
+        const value = await window.grecaptcha.execute(siteKey, {
+          action: recaptchaAction,
+        });
+        callback?.setResult(value);
+      });
+    }
+  }
+  $: {
+    // if recaptcha Action is 0 length, then it has not been passed in as a prop
+    // and we should use the journey tree value which we may have to wait for since
+    // it is not going to be available on mount immediatly due to be an async function
+    recaptchaAction = recaptchaAction.length === 0 ? journey?.tree ?? '' : recaptchaAction;
+    if (recaptchaAction.length) {
+      executeCaptcha();
+    }
```

Comment:
I think some of this can be cleaned up by directly using the actual journey stack store. We can then `subscribe` to the store and operate on the value it emits, which should be the whole stack. Then, you can iterate on that stack to find the journey name value. You could even make a derived store out of that store to give you the reactivity you probably want.